lib/elixir/test/elixir/code_test.exs

```diff
@@ -118,6 +118,11 @@ defmodule CodeTest do
       assert string_to_quoted.("foo + bar") == {:ok, {:+, [line: 1, column: 5], [foo, bar]}}
     end
 
+    test "returns an error tuple on hex errors" do
+      assert Code.string_to_quoted("\"\\x\"") ==
+               {:error, "missing hex sequence after \\x, expected \\xHH"}
```

Comment:
This is changing the contract of `Code.string_to_quoted` which returns errors in the format `{error, Tuple}`. So let's proceed as [described here](https://github.com/elixir-lang/elixir/pull/7809/files#r198758547) so we can keep all the error formats the same!